lib/core/resources/result.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:superformula_test/core/resources/failure.dart';
+
+abstract class Result<T> {
```

Comment:
**praise**: I really like this approach 😄 

---

On the technical side, you could use `const` constructors more and combine them with redirecting factories. Something like this may be cleaner:

```dart
abstract class Result<T> {
  const factory Result.failure(T f) = FailureResult;
  const factory Result.success(T data) = SuccessResult;
}

class SuccessResult<T> implements Result<T> {
  final T something;
  const SuccessResult(this.something);
}

class FailureResult<T> implements Result<T> {
  final T something;
  const FailureResult(this.something);
}
```

This is just a side-note however, nothing to worry about 🙂

Reply:
Awesome idea! Thanks for sharing!